packages/cactus-common/src/test/typescript/unit/objects/get-all-method-names.ts

```diff
@@ -1,48 +1,15 @@
-// tslint:disable: max-classes-per-file
 // tslint:disable-next-line: no-var-requires
-const tap = require("tap");
-import { Objects } from "../../../../main/typescript/public-api";
-
-class Base {
-  private readonly y: string;
-
-  constructor() {
-    this.y = "y";
-  }
-
-  getX() {
-    return "x";
-  }
-}
+import test from "tape";
 
-class A extends Base {
-  private readonly b: string;
-
-  constructor() {
-    super();
-    this.b = "b";
-  }
-
-  getA() {
-    return "a";
-  }
-}
+import { Objects } from "../../../../main/typescript/public-api";
+import { A } from "../../fixtures/dummy-classes";
 
-tap.test("handles inheritance correctly", (assert: any) => {
+test("handles inheritance correctly", (assert: any) => {
```

Comment:
The `assert` parameter can be typed by importing the `Test` type from the `tape` library using the brackets syntax like this:
```typescript
import test, { Test } from "tape";
```

If you do this then the IDE will have auto complete for all the possible assertion methods that the testing library provides (big help when you don't have to look up stuff like how do they call the deepEquals assertion method and other things like that.

Reply:
Update parameter type.